sdk/java/pulumi/src/test/java/com/pulumi/deployment/MocksTest.java

```diff
@@ -278,63 +277,53 @@ public static void myStack(Context ctx) {
     public static class MyMocks implements Mocks {
 
         @Override
-        public CompletableFuture<Tuples.Tuple2<Optional<String>, Object>> newResourceAsync(MockResourceArgs args) {
+        public CompletableFuture<ResourceResult> newResourceAsync(ResourceArgs args) {
             requireNonNull(args.type);
             switch (args.type) {
                 case "aws:ec2/instance:Instance":
                     return CompletableFuture.completedFuture(
-                            Tuples.of(Optional.of("i-1234567890abcdef0"), ImmutableMap.of("publicIp", "203.0.113.12"))
+                            ResourceResult.of(Optional.of("i-1234567890abcdef0"), ImmutableMap.of("publicIp", "203.0.113.12"))
```

Comment:
Perhaps a builder for ResourceResult? Builders never end it seems, but it could make this code more readable.

Reply:
my thinking was that a factory method is slightly preferred since there is only 2 args and it forces setting of the required parameter in compile time

Reply:
It just doesn't name the args, and their type is `Optional<String>` for example, no way to say it's an ID. But the docs probably say that. Just a nit, feel free to ignore.